File: app/src/main/java/ru/startandroid/develop/retrofittask/MessageViewModel.java
package ru.startandroid.develop.retrofittask;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

public class MessageViewModel extends AndroidViewModel {
    private MessageRepository messageRepository;
    private LiveData<List<MessageEntity>> messages;

    public MessageViewModel(@NonNull Application application) {
        super(application);
        this.messageRepository = new MessageRepository(application);
        messages = messageRepository.getMessage();
    }

    public void insertMessage(MessageEntity messageEntity){
        messageRepository.insertMessage(messageEntity);
    }

    public LiveData<List<MessageEntity>> getMessage(){
        return messages;
    }

}

File: app/src/main/java/ru/startandroid/develop/retrofittask/MessageViewModel.kt
package ru.startandroid.develop.retrofittask

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.viewModelScope
import kotlinx.coroutines.Dispatchers.IO
import kotlinx.coroutines.launch

class MessageViewModel(application: Application) : AndroidViewModel(application) {
    private val messageRepository: MessageRepository = MessageRepository(application)
    val message: LiveData<List<MessageEntity>> = messageRepository.message

    fun insertMessage() {
        viewModelScope.launch(IO) {
                messageRepository.fetchAndInsertData()
        }
    }

}